controllers/ocsinitialization/ocsoperator_config.go

```diff
@@ -0,0 +1,142 @@
+package ocsinitialization
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// This configmap is purely for the OCS operator to use
+	ocsOperatorConfigName = "ocs-operator-config"
+)
+
+// ensureOCSOperatorConfig ensures that the OCS operator config is present in the
+// cluster & has the correct values
+func (r *OCSInitializationReconciler) ensureOCSOperatorConfig(initialData *ocsv1.OCSInitialization) error {
+	ocsOperatorConfig := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ocsOperatorConfigName, Namespace: initialData.Namespace}, ocsOperatorConfig)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// If it does not exist, create the ConfigMap
+			return r.Client.Create(context.TODO(), newOCSOperatorConfig(initialData.Namespace))
+		}
+		return err
+	}
+	resourceVersion := ocsOperatorConfig.ResourceVersion
+
+	//Adding Ownerreference to the configmap if it's not present
+	ownerRefs := ocsOperatorConfig.GetOwnerReferences()
+	if len(ownerRefs) == 0 {
+		err = controllerutil.SetOwnerReference(initialData, ocsOperatorConfig, r.Client.Scheme())
+		if err != nil {
+			return err
+		}
+		err = r.Client.Update(context.TODO(), ocsOperatorConfig)
+		if err != nil {
+			return err
+		}
+	}
+	// Keep the clustername in sync with the cluster ID
+	err = r.updateClusterName(ocsOperatorConfig)
+	if err != nil {
+		return err
+	}
+	// restart the rook-ceph-operator pod if any change has occured to the configmap
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: ocsOperatorConfigName, Namespace: initialData.Namespace}, ocsOperatorConfig)
+	if err != nil {
+		return err
+	}
+	if resourceVersion != ocsOperatorConfig.ResourceVersion {
+		podList := &corev1.PodList{}
+		opts := []client.ListOption{
+			client.InNamespace(initialData.Namespace),
+			client.MatchingLabels(map[string]string{"key": "value"}),
+		}
+		err = r.Client.List(context.TODO(), podList, opts...)
+		if err != nil {
+			r.Log.Error(err, "Failed to list rook-ceph-operator pods")
+			return nil
+		}
+		err = r.Client.Delete(context.TODO(), &podList.Items[0])
```

Comment:
Restarting the rook operator pod should not be necessary. Rook has a watcher on this configmap and it will update the CSI driver every time the configmap is updated. So I believe we're already covered to pick up these settings in the configmap if they change.

Reply:
But Travis this one is not the rook-ceph-operator configmap, this is a new CM that we are introducing. Rook-ceph-operator will set an env variable by looking here, so I believe unless restarted the new value for env var won't come.